### PR TITLE
Sort shares by title and then creation date

### LINKED ISSUE #36906: Sort shares by title and then creation date
image

Currently they are not sorted at all
https://github.com/nextcloud/server/blob/078203afeab667cbe1ffbcfa76a635e1bf59b9d3/apps/files_sharing/src/views/SharingTab.vue#L264-L265

### PR DESCRIPTION
**Pull request by SamantaTarun**
* Resolves: https://github.com/nextcloud/server/issues/36906

Please don't close the PR. Let me know if any changes are required.

## Summary

Let me know if any changes are required.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_sharing/src/views/SharingTab.vue [hunk 1]
```diff
@@ -259,6 +259,7 @@ export default {
 				// create Share objects and sort by newest
 				const shares = data.ocs.data
 					.map(share => new Share(share))
+					.sort((a, b) => a.title > b.title)
 					.sort((a, b) => b.createdTime - a.createdTime)
```
Comment (skjnldsv) [inline apps/files_sharing/src/views/SharingTab.vue:263]: the last sort will cancel the previous one, you have to create a sorting function and fallback to `createdTime` if the `title` comparison is 0.
Also, you cannot compare strings with `>`. Use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

File: apps/files_sharing/src/views/SharingTab.vue [hunk 2]
```diff
@@ -259,6 +259,7 @@ export default {
 				// create Share objects and sort by newest
 				const shares = data.ocs.data
 					.map(share => new Share(share))
+					.sort((a, b) => a.title > b.title)
```
Comment (skjnldsv) [inline apps/files_sharing/src/views/SharingTab.vue:262]: the last sort will cancel the previous one, you have to create a sorting function and fallback to `createdTime` if the `title` comparison is 0.
Also, you cannot compare strings with `>`. Use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

**Review by skjnldsv — CHANGES_REQUESTED** [review]
See

File: apps/files_sharing/src/views/SharingTab.vue (hunk 2 above)
Comment (skjnldsv) [reply apps/files_sharing/src/views/SharingTab.vue]: @tarunsamanta2k20 we can properly review and help you if you cannot at realistically reproduce this PR.
We cannot accept fixes that are made by head only and not being a true representation of your local environnement.
There are tons of accessible ressources on how to setup a quick and easy Nextcloud (just cloning and using php's internal webserver would be enough). 
I sent you ressources about or requirements, and I see you haven't signed the DCO (see status checks at the bottom) :)

I appreciate the efforts, but somethings is telling me you haven't even read the checklist at the bottom of your main PR post :/

**Comment by tarunsamanta2k20** [comment]
@skjnldsv could you please test it once?

**Comment by skjnldsv** [comment]
> @skjnldsv could you please test it once?

See comment https://github.com/nextcloud/server/pull/36933#discussion_r1138356685

**Comment by szaimen** [comment]
@tarunsamanta2k20 I would recommend https://github.com/szaimen/nextcloud-easy-test for testing the changes you've made.

If you've installed docker locally, you should be able to test it with 
```
docker run -it \
-e SERVER_BRANCH=tarunsamanta2k20:tarun_samanta/#36906 \
-e COMPILE_SERVER=1 \
--name nextcloud-easy-test \
-p 127.0.0.1:8443:443 \
--volume="nextcloud_easy_test_npm_cache_volume:/var/www/.npm" \
ghcr.io/szaimen/nextcloud-easy-test:latest
```
More instructions are here: https://github.com/szaimen/nextcloud-easy-test#nextcloud-easy-test-instance

**Comment by tarunsamanta2k20** [comment]
@szaimen let me try once.

**Comment by tarunsamanta2k20** [comment]
@szaimen someone I managed to run the application but tell me how to add some shares. Because share tab is empty right now.

**Comment by szaimen** [comment]
> someone I managed to run the application but tell me how to add some shares.

Simply create a few users and then share to them and also create link shares?

**Comment by tarunsamanta2k20** [comment]
@szaimen let see tomorrow.

**Comment by tarunsamanta2k20** [comment]
@skjnldsv please check my recent commit, i have tested it, it's working fine.

**Comment by skjnldsv** [comment]
@tarunsamanta2k20 I'm sorry, but I cannot review or accept this PR
- You keep adding more commits instead of amending/squashing
- You are not writing commits descriptions
- You did not test the work you've been putting 
- You didn't even sign-off your work despite me linking our [contributing guidelines](https://github.com/nextcloud/server/blob/master/.github/CONTRIBUTING.md) and it being written down

This is too light as a contribution in my opinion :confused: 
I'm being straightforward, but you seem to have been contributing to Github repositories for multiple months now, so I assume you know how contributing works in general :)
I'm removing myself from reviewer, I don't have time to guide further. I will let others from the community take over :+1:

**Comment by tarunsamanta2k20** [comment]
> @tarunsamanta2k20 I'm sorry, but I cannot review or accept this PR
> 
> * You keep adding more commits instead of amending/squashing
> * You are not writing commits descriptions
> * You did not test the work you've been putting
> * You didn't even sign-off your work despite me linking our [contributing guidelines](https://github.com/nextcloud/server/blob/master/.github/CONTRIBUTING.md) and it being written down
> 
> This is too light as a contribution in my opinion 😕 I'm being straightforward, but you seem to have been contributing to Github repositories for multiple months now, so I assume you know how contributing works in general :) I'm removing myself from reviewer, I don't have time to guide further. I will let others from the community take over 👍
@skjnldsv 
Shall I modify this PR or create a new PR?

**Comment by tarunsamanta2k20** [comment]
@skjnldsv shall i create a new fresh PR?

**Comment by tarunsamanta2k20** [comment]
@skjnldsv please delete this PR. Don't close it. Just delete it. i am opening a fresh PR for this issue.

**Pull request closed without merging**